tests/func/test_ignore.py

```diff
@@ -1,186 +1,121 @@
-    repo_dir.create(
-        ignore_file,
-        "\n".join(
-            [to_posixpath(repo_dir.DATA_SUB), to_posixpath(new_data_path)]
-        ),
-    )
+def test_rename_file(tmp_dir, dvc):
+    tmp_dir.gen({"dir": {"foo": "foo", "bar": "bar"}})
+    mtime, size = get_mtime_and_size("dir", dvc.dvcignore)
 
-    mtime_sig, size = get_mtime_and_size(repo_dir.DATA_DIR, dvc_repo.dvcignore)
+    shutil.move("dir/foo", "dir/foo_new")
+    new_mtime, new_size = get_mtime_and_size("dir", dvc.dvcignore)
 
-    shutil.move(repo_dir.DATA_SUB, new_data_path)
+    assert new_mtime != mtime and new_size == size
 
-    new_mtime_sig, new_size = get_mtime_and_size(
-        repo_dir.DATA_DIR, dvc_repo.dvcignore
-    )
 
-    assert new_mtime_sig == mtime_sig
-    assert new_size == size
+def test_remove_ignored_file(tmp_dir, dvc):
+    tmp_dir.gen({"dir": {"ignored": "...", "other": "text"}})
+    tmp_dir.gen(DvcIgnore.DVCIGNORE_FILE, "dir/ignored")
 
+    mtime, size = get_mtime_and_size("dir", dvc.dvcignore)
 
-def test_mtime_changed_when_moving_non_ignored_file(dvc_repo, repo_dir):
-    new_data_path = repo_dir.DATA_SUB + "_new"
-    mtime, size = get_mtime_and_size(repo_dir.DATA_DIR, dvc_repo.dvcignore)
+    os.remove("dir/ignored")
+    new_mtime, new_size = get_mtime_and_size("dir", dvc.dvcignore)
 
-    shutil.move(repo_dir.DATA_SUB, new_data_path)
-    new_mtime, new_size = get_mtime_and_size(
-        repo_dir.DATA_DIR, dvc_repo.dvcignore
-    )
+    assert new_mtime == mtime and new_size == size
 
-    assert new_mtime != mtime
-    assert new_size == size
 
+def test_remove_file(tmp_dir, dvc):
+    tmp_dir.gen({"dir": {"foo": "foo", "bar": "bar"}})
+    mtime, size = get_mtime_and_size("dir", dvc.dvcignore)
 
-def test_metadata_unchanged_on_ignored_file_deletion(dvc_repo, repo_dir):
-    ignore_file = os.path.join(dvc_repo.root_dir, DvcIgnore.DVCIGNORE_FILE)
-    repo_dir.create(ignore_file, to_posixpath(repo_dir.DATA_SUB))
+    os.remove("dir/foo")
+    new_mtime, new_size = get_mtime_and_size("dir", dvc.dvcignore)
 
-    mtime, size = get_mtime_and_size(repo_dir.DATA_DIR, dvc_repo.dvcignore)
+    assert new_mtime != mtime and new_size != size
 
-    os.remove(repo_dir.DATA_SUB)
-    new_mtime, new_size = get_mtime_and_size(
-        repo_dir.DATA_DIR, dvc_repo.dvcignore
-    )
 
-    assert new_mtime == mtime
-    assert new_size == size
+def test_dvcignore_in_out_dir(tmp_dir, dvc):
+    tmp_dir.gen({"dir": {"foo": "foo", DvcIgnore.DVCIGNORE_FILE: ""}})
 
+    with pytest.raises(DvcIgnoreInCollectedDirError):
+        dvc.add("dir")
 
-def test_metadata_changed_on_non_ignored_file_deletion(dvc_repo, repo_dir):
-    mtime, size = get_mtime_and_size(repo_dir.DATA_DIR, dvc_repo.dvcignore)
 
-    os.remove(repo_dir.DATA_SUB)
-    new_mtime_sig, new_size = get_mtime_and_size(
-        repo_dir.DATA_DIR, dvc_repo.dvcignore
-    )
+# @efiop why do we need to parametrize this?
```

Comment:
@efiop ^

Reply:
@Suor @mroutis because we need to ensure that subdirs work too. Or am I missing something?

Reply:
Testing that `.dvcignore` works for subdirs and testing that ignored `.dvcignore` files won't be collected looks like two different things. I am not strong on this though, so if you think this is ok, then resolve it.

Reply:
Yeah, could be better. Let's keep it for now though, please feel free to create an issue.

P.S. merged before removing this comment, will fix upstream.